Battleship/GridTile.cs

```diff
@@ -46,6 +46,11 @@ public bool HasShip
                 return Ship is not null;
             }
         }
+        /// <summary>
+        /// Once the tile has been shot, prevent it from being shot again.
+        /// Set by the Grid object that owns this instance.
+        /// </summary>
+        public bool IsShot = false;
```

Comment:
```suggestion
        public bool IsShot { get; set;} = false;
```

make this a property